modules/imgcodecs/test/test_tiff.cpp

```diff
@@ -46,6 +46,201 @@ TEST(Imgcodecs_Tiff, decode_tile16384x16384)
     EXPECT_EQ(0, remove(file4.c_str()));
 }
 
+#ifdef __ANDROID__
+// Test disabled as it uses a lot of memory.
+// It is killed with SIGKILL by out of memory killer.
+TEST(Imgcodecs_Tiff, DISABLED_decode_issue22388_16UC1)
+#else
+TEST(Imgcodecs_Tiff, decode_issue22388_16UC1)
+#endif
+{
```

Comment:
Please add `CV_TEST_TAG_MEMORY_6GB`.
(problem is not android only - 32-bit environments may fail too, e.g. [here](http://pullrequest.opencv.org/buildbot/builders/precommit_windows32/builds/100061)).

Also these tests take some extra time, so `CV_TEST_TAG_LONG` should be added too.

As a first line of tests add this:

```
applyTestTag(
    CV_TEST_TAG_MEMORY_6GB,
    CV_TEST_TAG_LONG
);
```

Reply:
Thank you for your review. 

Improved so that a test tag according to the required amount of memory is added before performing an image loading test. 

So that, tests are no longer impacted based on differences in compilation environments.